go/inst/instance_dao.go

```
@@ -3068,6 +3155,79 @@ func FigureInstanceKey(instanceKey *InstanceKey, thisInstanceKey *InstanceKey) (
 	return figuredKey, nil
 }
 
+// Obtain information about Group Replication  for this host as well as other hosts who are members of the same group
+// (if any). This method swallows up any errors, interpreting them as if the host does not support group replication
+// (e.g. because the group replication plugin is not loaded, therefore the corresponding performance schema tables and
+// global variables are not known to the server.
+func PopulateGroupReplicationInformation(instance *Instance, db *sql.DB) {
```

I'm not sure ignoring the error is the best way here, I catch error and ignore only for expected errors if there is no group replication plugin, etc, but not ignore all errors

---

Agree

---

Is it not possible to validate whether this host at all supports group replication, and based on that finding to invoke/skip this function?

---

Yes, we can check whether the plugin is installed and active, but this requires an extra query. I will modify this to check the error codes from the query and only ignore them if they are either because the group replication global variables or performance schema tables are not known, which would mean that the host does not support group replication.

---

Done.